src/snapshot.js

```diff
@@ -69,6 +69,23 @@ export function getPlayerSnapshot(player) {
   }
   snapshotObject.suppressedTracks = suppressedTracks;
 
+  // iOS Safari will change caption mode to 'showing' if a user previously
+  // turned captions on manually for that video source, so this will re-disable
+  // them in case that occurs during ad playback
+  if (videojs.browser.IS_IOS && !Array.isArray(tracks)) {
+    tracks.one('change', function(event) {
+      if (player.ads.state === 'ad-playback') {
+        const trackList = event.target.tracks_;
```

Comment:
You can use a regular old forloop or you can delegate it to the forEach method.
```js
Array.prototype.forEach(player.textTracks(), track => track.mode === 'showing' ? track.mode = 'disabled' : '');
```

(except don't actually use the ternary there :P)